src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/caseworker/TransferCaseCallbackHandler.java

```
@@ -0,0 +1,120 @@
+
+    @Override
+    public List<CaseEvent> handledEvents() {
+        return EVENTS;
+    }
+
+    @Override
+    public List<Role> getSupportedRoles() {
+        return ROLES;
+    }
+
+    private CallbackResponse prepopulateData(CallbackParams callbackParams) {
+
+     return AboutToStartOrSubmitCallbackResponse
+            .builder()
+            .data(ImmutableMap.of(
+                "dateOfTransfer", "TODAYS_DATE", // TODO
+                "reasonForTransfer",null,
+                "nameOfTransferCourt", null,
+                "addressOfTransferCourt", null
+            ))
+            .build();
+    }
+
+    private CallbackResponse performBulkPrintTransfer(CallbackParams callbackParams) {
+
+        CCDCase ccdCase = caseDetailsConverter.extractCCDCase(callbackParams.getRequest().getCaseDetails());
+
+        try {
+
+            var noticeOfTransferLetter = bulkPrintTransferService.generateNoticeOfTransferLetter(ccdCase);
+
+            bulkPrintTransferService.addNoticeOfTransferToCaseRecordToCaseDocuments(ccdCase, noticeOfTransferLetter);
+
+            if (!isDefendantLinked(ccdCase)) {
+                var transferNoticeLetter = bulkPrintTransferService.generateTransferNoticeLetter(ccdCase);
+
+                bulkPrintTransferService.addTransferLetterNoticeToCaseDocuments(ccdCase, transferNoticeLetter);
+            }
+
+            bulkPrintTransferService.sendCaseDocumentsToBulkPrint(ccdCase);
+
+            bulkPrintTransferService.sendClaimUpdatedEmailToClaimant(ccdCase);
+
+            if (isDefendantLinked(ccdCase)) {
+                bulkPrintTransferService.sendClaimUpdatedEmailToDefendant(ccdCase);
+            }
+
+        } catch (Exception e) {
+            return AboutToStartOrSubmitCallbackResponse
+                .builder()
+                .errors(Collections.singletonList(
+                    "There was a technical problem. Some items may not have been sent. You may want to try again."))
+                .build();
+        }
```

Probably best not to swallow this exception. CCD will display its own generic error message when there are exceptions in callbacks.

---

I based the error handling on what I saw in GeneralLetterCallbackHandler.java. Please can you give me an example of the best practice.

---

I would say that best practice would be to just allow the exception to cascade up. It would then report in the app insights monitoring and CCD will report a generic error message.

---

Fixed in commit 4